main.cpp

```diff
@@ -50,8 +51,51 @@ struct Wrapper
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")\n";
+    }
+private:
+    Type val;
+};
+
+// 8) Wrapper class specialization for the Point class
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& t) : val(std::move(t)) 
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val.toString() << ")\n";
+    }
+private:
+    Point val;
+};
+
+
+// 4) single-parameter version of template
+template<typename T>
+void variadicHelper(T&& first)
+{
+    Wrapper<T>( std::forward<T>(first) ).print();
+}
+
+// 3) recursive variadic template
+template<typename T, typename ... Args>
+void variadicHelper(T&& first, Args&& ... everythingElse)
+{
+    Wrapper<T>( std::forward<T>(first) ).print();
+
+    variadicHelper( std::forward<Args>(everythingElse) ... ); // recursive call
+}
```

Comment:
Project9.1 challenge 2:
There is another recursive solution.  it is possible for `Args ...` to be empty.
your task: remove your implementation for instruction `4)` and implement this alternate solution.
implement it **after** your existing `variadicHelper` function.
hint: what does the **recursive call** actually look like when called with an empty Variadic Parameter list?